### ISSUE TITLE
Undefined variable false positive with unreachable expression

### ISSUE THREAD

**Issue by hauntsaninja** [microsoft/pylance-release#452]
-   Language Server version: Pylance language server 2020.9.8 (pyright a0620921)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8

<img width="705" alt="Screen Shot 2020-10-03 at 12 36 40 AM" src="https://user-images.githubusercontent.com/12621235/94986134-09af5300-0511-11eb-9f33-502114da8a47.png">

## Expected behaviour

Pylance just complains about the unreachable (given my Python version) code 

## Actual behaviour

Pylance complains about `message` being undefined (in addition to the unreachable code)

## Code Snippet / Additional information

```python
import sys

def hello() -> None:
    message = "asdf"
    if sys.version_info < (3, 5) and message:
        print("hi")
```

**Comment by erictraut** [comment]
Thanks for the bug report. This was related to reachability analysis. The control flow node represented by `message` is provably never reachable within the function's body because the first part of the condition expression statically evaluates to False. This caused the analyzer to assume incorrectly that `message` was unbound.

I've added a fix, and it will be in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.10.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020100-7-october-2020
